### ISSUE TITLE
the bug of gr.FileExplorer

### ISSUE THREAD

**Issue by Weishaoya** [gradio-app/gradio#6542]
### Describe the bug

dataset_selection = gr.Dropdown(value=DirName.PUBLIC_DATASET_DIR,
                                            label='数据集',
                                            info='选择已有数据集',
                                            choices=list_children_folders(USER_DATASET_DIR),
                                            interactive=True)              
 file_explorer = gr.FileExplorer(root=USER_DATASET_DIR, interactive=True)
 dataset_selection.select(
        lambda x: gr.update(root=os.path.join(USER_DATASET_DIR, x)), inputs=dataset_selection,
        outputs=file_explorer)
The code above, gr.Dropdown is unable to update gr.FileExplorer.

### Have you searched existing issues?  🔎

- [X] I have searched and found no existing issues

### Reproduction

```python
dataset_selection = gr.Dropdown(value=DirName.PUBLIC_DATASET_DIR,
label='数据集',
info='选择已有数据集',
choices=list_children_folders(USER_DATASET_DIR),
interactive=True)
file_explorer = gr.FileExplorer(root=USER_DATASET_DIR, interactive=True)
dataset_selection.select(
lambda x: gr.update(root=os.path.join(USER_DATASET_DIR, x)), inputs=dataset_selection,
outputs=file_explorer)
```

The code above, gr.Dropdown is unable to update gr.FileExplorer.

### System Info

```shell
Flask                     2.3.3
Flask-Cors                4.0.0
fonttools                 4.44.0
frozenlist                1.4.0
fsspec                    2023.10.0
gradio                    4.5.0
gradio_client             0.7.0
h11                       0.14.0
```

### Severity

I can work around it

**Comment by Weishaoya** [comment]
I am using Python 3.11, and the bug still persists. I hope you can update the version of gradio and solve the bug. Thanks!

**Comment by cosmosof** [comment]
@freddyaboulton is this fix released on `gradio==4.7.1` I am running into the same issue.

**Comment by freddyaboulton** [comment]
Will be released in the next release!
